File: test/f79/leave.fs
t{ : gd5 123 swap 0 do
        i 4 > if drop 234 leave then
    loop ; -> }t

t{ 1 gd5 -> 123 }t
t{ 5 gd5 -> 123 }t
t{ 6 gd5 -> 234 }t

File: primitive/f79/core/leave.c
pr_leave:
{
    /* LEAVE ( -- )

       Force termination of a DO-LOOP at the next LOOP or +LOOP by
       setting the loop limit equal to the current value of the
       index. The index itself remains unchanged, and execution
       proceeds normally until the loop terminating word is
       encountered.
     */

    *(rp+1) = *rp;
}
